test/parallel/xyk-pallet.API.parachainStacking.joinCandidate.test.ts

```diff
@@ -0,0 +1,236 @@
+    liqToken,
+    minStk.muln(2)
+  );
+  expect(result.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  const isUserInCandidate = await Staking.isUserInCandidateList(
+    testUser1.keyRingPair.address
+  );
+  expect(isUserInCandidate).toBeTruthy();
+});
+
+test("A user can schedule and execute bond more", async () => {
+  const api = getApi();
+
+  testUser1.addAsset(liqToken);
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  const userAmountBeforeJoining =
+    testUser1.getAsset(liqToken)?.amountBefore.reserved!;
+
+  await joinCandidate(testUser1.keyRingPair, liqToken, minStk.muln(2)).then(
+    (result) => {
+      expect(result.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    }
+  );
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  const userAmountBeforeExecuting =
+    testUser1.getAsset(liqToken)?.amountBefore.reserved!;
+
+  await signTx(
+    api,
+    Staking.scheduleCandidateBondMore(minStk),
+    testUser1.keyRingPair
+  );
+
+  await waitNewStakingRound();
+  await waitNewStakingRound();
+
+  await signTx(
+    api,
+    Staking.executeBondRequest(testUser1),
+    testUser1.keyRingPair
+  );
+
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const userAmountAfterExecuting =
+    testUser1.getAsset(liqToken)?.amountAfter.reserved!;
+
+  expect(userAmountBeforeJoining).bnEqual(BN_ZERO);
+  expect(userAmountBeforeExecuting).bnEqual(minStk.muln(2));
+  expect(userAmountAfterExecuting).bnEqual(minStk.muln(3));
+});
+
+test("A user can delegate more using liq token", async () => {
+  const api = getApi();
+
+  const testUser2 = new User(keyring);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(testUser2, minStk.muln(1000)),
+    Assets.mintToken(liqToken, testUser2, minStk.muln(2))
+  );
+
+  testUser1.addAsset(liqToken);
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await joinCandidate(testUser1.keyRingPair, liqToken, minStk.muln(2)).then(
+    (result) => {
+      expect(result.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    }
+  );
+
+  const collatorAmountAfterJoining = await getUserAmount(testUser1, liqToken);
+
+  await delegate(
+    testUser2.keyRingPair,
+    liqToken,
+    minStk.divn(2),
+    "availablebalance"
+  );
+
+  await signTx(
+    api,
+    Staking.scheduleDelegatorBondMore(testUser1, minStk.divn(2)),
+    testUser2.keyRingPair
+  );
+
+  const collatorAmountBeforeExecuting = await getUserAmount(
+    testUser1,
+    liqToken
+  );
+
+  await waitNewStakingRound();
+  await waitNewStakingRound();
+
+  await signTx(
+    api,
+    Staking.executeDelegationRequest(testUser2, testUser1),
+    testUser2.keyRingPair
+  );
+
+  const collatorAmountAfterExecuting = await getUserAmount(testUser1, liqToken);
+
+  expect(collatorAmountAfterJoining).bnEqual(minStk.muln(2));
+  expect(collatorAmountBeforeExecuting).bnEqual(
+    minStk.muln(2).add(minStk.divn(2))
+  );
+  expect(collatorAmountAfterExecuting).bnEqual(minStk.muln(3));
```

Comment:
same as before, check the parachainStaking storage items.

Reply:
Done

Reply:
??

Reply:
Done